code/modules/mob/living/simple_animal/parrot.dm

```diff
@@ -923,7 +923,7 @@
 	S["longestsurvival"]	>> longest_survival
 	S["longestdeathstreak"] >> longest_deathstreak
 
-	if(isnull(speech_buffer))
+	if(!islist(speech_buffer))
 		speech_buffer = list()
 
 /mob/living/simple_animal/parrot/Poly/proc/Write_Memory()

```

Comment:
You also need to make it not save if its been fucked with, what for not fucking up the rest of the saved data.